field/src/types.rs

```diff
@@ -341,6 +341,26 @@ pub trait Field:
     /// Returns `n % Self::characteristic()`.
     fn from_noncanonical_u128(n: u128) -> Self;
 
+    /// Returns `x % Self::CHARACTERISTIC`.
+    ///
+    /// Implemented by default via `from_noncanonical_u128`
+    /// Override, if you have a faster implementation for your field.
+    fn from_noncanonical_u64(n: u64) -> Self {
+        Self::from_noncanonical_u128(n as u128)
+    }
+
+    /// Returns `n` as an element of this field.
+    ///
+    /// Implemented by default via `from_noncanonical_u128` and case analysis.
+    /// Override, if you have a faster implementation for your field.
+    fn from_noncanonical_i64(n: i64) -> Self {
+        if n < 0 {
+            -Self::from_noncanonical_u128((n as i128).unsigned_abs())
```

Comment:
sorry , how to understand this negation?

Reply:
I removed the default implementation.